apps/web-staking/src/app/components/ui/modals/BaseModal.tsx

```diff
@@ -56,10 +113,12 @@ const BaseModal = ({
             </span>}
             <span className="block font-bold text-white text-2xl mb-[18px]">{modalHeader}</span>
             <span className="block text-[17px] font-medium text-americanSilver">{modalBody}</span>
+            {isDropdown && <SearchableDropdown dropdownOptionsCount={countries.length} isOpen={isOpen!} selectedValue={selectedCountry!} selectedValueRender={selectedCountry!} setSelectedValue={setSelectedCountry!} setIsOpen={setIsOpen!} getDropdownItems={() => countries} extraClasses={{ dropdown: "!w-full my-3", dropdownOptions: "!w-full" }} isInvalid={isError}  />}
+            {isError && <span className="block text-lg font-medium text-[#F76808]">{errorMessage}</span>}
             <div className="flex justify-end items-end mt-2">
               {!withOutCancelButton &&
                 <TextButton buttonText={cancelText} onClick={closeModal} textClassName="!text-lg !font-bold" />}
-              <PrimaryButton onClick={onSubmit} btnText={submitText} size="sm" isDisabled={isDisabled} />
+              <PrimaryButton onClick={isDropdown ? onClickHelper : onSubmit} btnText={submitText} size="sm" isDisabled={isDisabled || isValidCountry()} />
```

Comment:
Should this be || !isValidCountry()?

Reply:
Function name was inverted and very misleading. Has been renamed.